app/commands/user_track/reset.rb

```diff
@@ -5,10 +5,16 @@ class UserTrack::Reset
   delegate :user, :track, to: :user_track
 
   def call
-    user_track.solutions.map(&:published_exercise_representation).compact.each do |representation|
-      Exercise::Representation::UpdatePublishedSolutions.defer(representation, wait: 10.seconds)
-    end
-    user_track.solutions.update_all("user_id = #{User::GHOST_USER_ID}, unique_key = UUID()")
+    # We need to get the published representation before we're updating the
+    # solutions, as at that point the link will be gone
+    published_exercise_representations = Exercise::Representation.where(
+      id: user_track.solutions.select(:published_exercise_representation_id)
+    ).to_a
+    user_track.solutions.update_all(%(
+      user_id = #{User::GHOST_USER_ID},
+      unique_key = UUID\(\),
+      published_exercise_representation_id = NULL
```

Comment:
This ensures that the solution won't be counted when updating `num_published_solutions` of the exercise representation